test/testsuppressions.cpp

```diff
@@ -401,6 +403,59 @@ class TestSuppressions : public TestFixture {
         ASSERT_EQUALS(true, s2.isSuppressed(errorMessage("abc", "include/1.h", 142)));
     }
 
+    void suppressionsLine0() {
+
+        const std::string filename("test.cpp");
+        std::map<std::string, std::string> files;
+        files.insert(std::make_pair(filename, ""));
+
+        std::map<std::string, std::string> tests;
+        tests.insert(std::make_pair("", "[test.cpp:0] -> [:0]: (error) syntax error\n"));
+        tests.insert(std::make_pair("syntaxError:*:0", ""));
+
+        for (auto it = tests.cbegin(); tests.cend() != it; ++it) {
+            errout.str("");
+
+            CppCheck cppCheck(*this, true);
+            Settings& settings = cppCheck.settings();
+            settings.exitCode = 1;
+            settings.inlineSuppressions = true;
+            settings.addEnabled("information");
+            settings.jointSuppressionReport = true;
+            if (it->first.size() > 0) {
+                std::string r = settings.nomsg.addSuppressionLine(it->first);
+                ASSERT_EQUALS("", r);
+            }
+            Tokenizer mTokenizer(&settings, this);
+
+            try {
+                mTokenizer.syntaxError(nullptr);
+            } catch (const InternalError& e) {
+                std::list<ErrorLogger::ErrorMessage::FileLocation> locationList;
+                ErrorLogger::ErrorMessage::FileLocation loc;
+                if (e.token) {
+                    loc.line = e.token->linenr();
+                    const std::string fixedpath = Path::toNativeSeparators(mTokenizer.list.file(e.token));
+                    loc.setfile(fixedpath);
+                } else {
+                    ErrorLogger::ErrorMessage::FileLocation loc2;
+                    loc2.setfile(Path::toNativeSeparators(filename));
+                    locationList.push_back(loc2);
+                    loc.setfile(mTokenizer.list.getSourceFilePath());
+                }
+                locationList.push_back(loc);
+                ErrorLogger::ErrorMessage errmsg(locationList,
+                                                 mTokenizer.list.getSourceFilePath(),
+                                                 Severity::error,
+                                                 e.errorMessage,
+                                                 e.id,
+                                                 false);
+                cppCheck.reportErr(errmsg);
```

Comment:
I am not sure if you really need to test this with the entire CppCheck and Tokenizer.

Can't you just write something like:

    void suppressionsLine0() {
        Suppressions suppressions;
        suppressions.addSuppressionLine("syntaxError:*:0");
        ASSERT_EQUALS(true, suppressions.isSuppressed(errorMessage("syntaxError", "test.cpp", 0)));
    }

Reply:
ah perhaps, this is my first contribution to cppcheck so I'm not very familiar with it yet :D perhaps your code works! I'll check it in a week after my holidays!

Reply:
ok!

Reply:
> this is my first contribution to cppcheck so I'm not very familiar with it yet

Yes of course! I totally understand this. Sorry, I meant no disrespect at all.

Reply:
I know you didn't ;) now that I see your proposal more clearly, it makes a lot of sense 👍